File: Components/ASP.NET/CS/ASPxVerticalGridDemos/App_Code/LargeDatabaseModel.cs
using System.Data.Entity;
using System.Data.Entity.ModelConfiguration;

public partial class LargeDatabaseContext : ContextBase {
    static LargeDatabaseContext() {
        Database.SetInitializer<LargeDatabaseContext>(null);
    }

    public LargeDatabaseContext() : base("LargeDatabaseConnectionString") { }

    public DbSet<Email> Emails { get; set; }
    public DbSet<Person> Persons { get; set; }

    protected override void OnModelCreating(DbModelBuilder modelBuilder) {
        modelBuilder.Configurations.Add(new EmailMap());
        modelBuilder.Configurations.Add(new PersonMap());
    }
}

public partial class Person {
    public int ID { get; set; }
    public string FirstName { get; set; }
    public string LastName { get; set; }
    public string Phone { get; set; }
}

public partial class Email {
    public int ID { get; set; }
    public string Subject { get; set; }
    public string From { get; set; }
    public System.DateTime Sent { get; set; }
    public long Size { get; set; }
    public bool HasAttachment { get; set; }
}


public class PersonMap : EntityTypeConfiguration<Person> {
    public PersonMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.FirstName)
            .IsRequired()
            .HasMaxLength(32);

        this.Property(t => t.LastName)
            .IsRequired()
            .HasMaxLength(32);

        this.Property(t => t.Phone)
            .IsRequired()
            .HasMaxLength(20);

        // Table & Column Mappings
        this.ToTable("Persons");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.FirstName).HasColumnName("FirstName");
        this.Property(t => t.LastName).HasColumnName("LastName");
        this.Property(t => t.Phone).HasColumnName("Phone");
    }
}

public class EmailMap : EntityTypeConfiguration<Email> {
    public EmailMap() {
        // Primary Key
        this.HasKey(t => t.ID);

        // Properties
        this.Property(t => t.Subject)
            .IsRequired()
            .HasMaxLength(100);

        this.Property(t => t.From)
            .IsRequired()
            .HasMaxLength(32);

        // Table & Column Mappings
        this.ToTable("Emails");
        this.Property(t => t.ID).HasColumnName("ID");
        this.Property(t => t.Subject).HasColumnName("Subject");
        this.Property(t => t.From).HasColumnName("From");
        this.Property(t => t.Sent).HasColumnName("Sent");
        this.Property(t => t.Size).HasColumnName("Size");
        this.Property(t => t.HasAttachment).HasColumnName("HasAttachment");
    }
}

File: Components/ASP.NET/VB/ASPxVerticalGridDemos/App_Code/LargeDatabaseModel.vb
Imports Microsoft.VisualBasic
Imports System.Data.Entity
Imports System.Data.Entity.ModelConfiguration

Partial Public Class LargeDatabaseContext
	Inherits ContextBase
	Shared Sub New()
		Database.SetInitializer(Of LargeDatabaseContext)(Nothing)
	End Sub

	Public Sub New()
		MyBase.New("LargeDatabaseConnectionString")
	End Sub

	Public Property Emails() As DbSet(Of Email)
	Public Property Persons() As DbSet(Of Person)

	Protected Overrides Sub OnModelCreating(ByVal modelBuilder As DbModelBuilder)
		modelBuilder.Configurations.Add(New EmailMap())
		modelBuilder.Configurations.Add(New PersonMap())
	End Sub
End Class

Partial Public Class Person
	Public Property ID() As Integer
	Public Property FirstName() As String
	Public Property LastName() As String
	Public Property Phone() As String
End Class

Partial Public Class Email
	Public Property ID() As Integer
	Public Property Subject() As String
	Public Property From() As String
	Public Property Sent() As System.DateTime
	Public Property Size() As Long
	Public Property HasAttachment() As Boolean
End Class


Public Class PersonMap
	Inherits EntityTypeConfiguration(Of Person)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.FirstName).IsRequired().HasMaxLength(32)

		Me.Property(Function(t) t.LastName).IsRequired().HasMaxLength(32)

		Me.Property(Function(t) t.Phone).IsRequired().HasMaxLength(20)

		' Table & Column Mappings
		Me.ToTable("Persons")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.FirstName).HasColumnName("FirstName")
		Me.Property(Function(t) t.LastName).HasColumnName("LastName")
		Me.Property(Function(t) t.Phone).HasColumnName("Phone")
	End Sub
End Class

Public Class EmailMap
	Inherits EntityTypeConfiguration(Of Email)
	Public Sub New()
		' Primary Key
		Me.HasKey(Function(t) t.ID)

		' Properties
		Me.Property(Function(t) t.Subject).IsRequired().HasMaxLength(100)

		Me.Property(Function(t) t.From).IsRequired().HasMaxLength(32)

		' Table & Column Mappings
		Me.ToTable("Emails")
		Me.Property(Function(t) t.ID).HasColumnName("ID")
		Me.Property(Function(t) t.Subject).HasColumnName("Subject")
		Me.Property(Function(t) t.From).HasColumnName("From")
		Me.Property(Function(t) t.Sent).HasColumnName("Sent")
		Me.Property(Function(t) t.Size).HasColumnName("Size")
		Me.Property(Function(t) t.HasAttachment).HasColumnName("HasAttachment")
	End Sub
End Class
